iox_query/src/provider.rs

```diff
@@ -2289,13 +2375,29 @@ mod test {
         ];
         assert_batches_sorted_eq!(&expected, &raw_data(&chunks).await);
 
-        let deduplicator = Deduplicater::new(IOxSessionContext::with_testing());
+        let deduplicator = Deduplicater::new(IOxSessionContext::with_testing(), false);
         let plan = deduplicator
-            .build_scan_plan(Arc::from("t"), schema, chunks, Predicate::default(), None)
+            .build_scan_plan(
+                Arc::from("t"),
+                Arc::clone(&schema),
+                chunks.clone(),
+                Predicate::default(),
+                None,
+            )
             .unwrap();
         let batch = test_collect(plan).await;
         // No duplicates so no sort at all. The data will stay in their original order
         assert_batches_sorted_eq!(&expected, &batch);
+
+        // ----------------------------------
+        // Test with disable_deduplication on
+        let deduplicator = Deduplicater::new(IOxSessionContext::with_testing(), true);
```

Comment:
same thing here, stylistically this might be easier to read if it was like

```suggestion
        let deduplicator = Deduplicater::new(IOxSessionContext::with_testing())
          .with_deduplication(false);
```